src/core/tests/type_prop/slice_scatter.cpp

```diff
@@ -0,0 +1,1055 @@
+    EXPECT_EQ(op->get_output_partial_shape(0), PartialShape({5, 5, 5}));
+}
+
+TEST_F(TypePropSliceScatterTest, default_ctor_axes) {
+    const auto data = std::make_shared<Parameter>(element::f32, Shape{5, 5, 5});
+    const auto updates = std::make_shared<Parameter>(element::f32, Shape{2, 5, 5});
+    const auto start = std::make_shared<Constant>(element::i32, Shape{2}, std::vector<int64_t>{0, 1});
+    const auto stop = std::make_shared<Constant>(element::i32, Shape{2}, std::vector<int64_t>{25, 5});
+    const auto step = std::make_shared<Constant>(element::i32, Shape{2}, std::vector<int64_t>{1, 2});
+    const auto axes = std::make_shared<Constant>(element::i32, Shape{2}, std::vector<int64_t>{2, 0});
+    const auto op = make_op();
+    op->set_arguments(OutputVector{data, updates, start, stop, step, axes});
+    op->validate_and_infer_types();
+    EXPECT_EQ(op->get_output_size(), 1);
+    EXPECT_EQ(op->get_input_size(), 6);
+    EXPECT_EQ(op->get_output_element_type(0), element::f32);
+    EXPECT_EQ(op->get_output_partial_shape(0), PartialShape({5, 5, 5}));
+}
+
+TEST_F(TypePropSliceScatterTest, basic_const_inputs) {
+    PartialShape data_shape{10, 10, 10, 10, 10, 10, 10, 10};
+    PartialShape updates_shape{7, 4, 10, 10, 9, 9, 5, 10};
+
+    std::vector<int32_t> start_val{1, 1, -20, 9, 9, 9, 9, 20};
+    std::vector<int32_t> stop_val{8, 8, 20, -11, 0, -10, -11, -20};
+    std::vector<int32_t> step_val{1, 2, 1, -1, -1, -1, -2, -1};
+
+    std::vector<int32_t> axes_val(start_val.size());
+    std::iota(axes_val.begin(), axes_val.end(), 0);
+
+    element::Type_t et = element::i32;
+
+    std::vector<std::vector<int32_t>> input_vals{start_val, stop_val, step_val, axes_val};
+    const auto op = make_op_const_inputs(input_vals, data_shape, updates_shape, et);
+
+    EXPECT_EQ(op->get_element_type(), et);
+    EXPECT_EQ(op->get_output_partial_shape(0), data_shape);
+}
+
+TEST_F(TypePropSliceScatterTest, basic_const_inputs_dif_steps) {
+    PartialShape data_shape{5, 5, 5, 5, 9, 9};
+    PartialShape updates_shape{3, 2, 2, 1, 5, 3};
+
+    std::vector<int32_t> start_val{0, 0, 0, 0, 0, 0};
+    std::vector<int32_t> stop_val{5, 5, 5, 5, 9, 9};
+    std::vector<int32_t> step_val{2, 3, 4, 5, 2, 3};
+
+    element::Type_t et = element::i32;
+
+    std::vector<std::vector<int32_t>> input_vals{start_val, stop_val, step_val};
+    const auto op = make_op_const_inputs(input_vals, data_shape, updates_shape, et);
+
+    EXPECT_EQ(op->get_element_type(), et);
+    EXPECT_EQ(op->get_output_partial_shape(0), data_shape);
+}
+
+TEST_F(TypePropSliceScatterTest, basic_const_inputs_dif_neg_steps) {
+    PartialShape data_shape{5, 5, 5, 5, 9, 9};
+    PartialShape updates_shape{2, 2, 1, 1, 4, 3};
+
+    std::vector<int32_t> start_val{5, 5, 5, 5, 9, 9};
+    std::vector<int32_t> stop_val{0, 0, 0, 0, 0, 0};
+    std::vector<int32_t> step_val{-2, -3, -4, -5, -2, -3};
+
+    element::Type_t et = element::i32;
```

Comment:
Use different types in different tests? Or only one is correct then add it to test fixture?